mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -656,7 +656,8 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
         itemFactors.setName(s"itemFactors-$iter").persist(intermediateRDDStorageLevel)
         // TODO: Generalize PeriodicGraphCheckpointer and use it here.
         if (shouldCheckpoint(iter)) {
-          itemFactors.checkpoint() // itemFactors gets materialized in computeFactors.
+          // itemFactors gets materialized in computeFactors & here.
+          ALS.checkpointAndCleanParents(itemFactors)
```

Comment:
@MLnick Actually, that plan doesn't work - checkpointing kills all of the parents information we need to clean up the shuffle files. I could refactor this so that we capture the dependency information needed - but a count() on a cached RDD should be low enough cost I'm not sure it would be worth it. What are your thoughts?